src/cascadia/TerminalControl/SearchBoxControl.cpp

```diff
@@ -155,12 +155,18 @@ namespace winrt::Microsoft::Terminal::Control::implementation
             // search box remains in Visible state (though not really *visible*) during the
             // first load. So, we only need to apply this check here (after checking that
             // we're done initializing).
-            if (Visibility() == Visibility::Visible)
+            if (IsOpen())
             {
                 callback();
                 return;
             }
 
+            // Stop ongoing close animation if any
+            if (CloseAnimation().GetCurrentState() == Media::Animation::ClockState::Active)
+            {
+                CloseAnimation().Stop();
+            }

```

Comment:
I'm not sure I understand how us previously ignoring the animation state would result in
> Fast closing and re-opening SearchBox would leave search highlights in an inconsistent state. 

Could you explain how that happens?

Reply:
When searchbox is in focus, if you press `Esc` and then quickly do a "find next/prev match" action, it will trigger this issue. The trick is to start a search when searchbox is closing. It's easier if you have find-match action binded to a single key, eg. <kbd>F3</kbd>

Reply:
Not to forget, highlights should be active (i.e. one or more search results available) when you close SB.